improver/calibration/ensemble_calibration.py

```diff
@@ -606,16 +622,8 @@ def create_coefficients_cube(self, optimised_coeffs, historic_forecast):
             )
             raise ValueError(msg)
 
-        coefficient_index = iris.coords.DimCoord(
-            np.arange(len(optimised_coeffs), dtype=np.int32),
-            long_name="coefficient_index",
-            units="1",
-        )
-        coefficient_name = iris.coords.AuxCoord(
-            coeff_names, long_name="coefficient_name", units="no_unit"
-        )
-        dim_coords_and_dims = [(coefficient_index, 0)]
-        aux_coords_and_dims = [(coefficient_name, 0)]
+        dim_coords_and_dims = []
+        aux_coords_and_dims = []
 
         # Create a forecast_reference_time coordinate.
         frt_point = cycletime_to_datetime(self.current_cycle)
```

Comment:
Second refactoring / simplification thought regarding time coordinates: lines 629-655 could be simplified substantially I am sure.  But I think some reasoning needs to be clarified here.  Some questions:
- Why is a forecast reference time coordinate being created with the current cycletime?  Is this ever used?  What happens if this is not created?
- What happens if the current cycle time is not the same time of day as the forecast reference time on the historic forecast cube(s)?
- The forecast period coordinate seems useful and valid.
- Creating a time coordinate from a calculated frt and correct fp seems dangerous.  The time coordinate - in particular the time of day - should be derived from the time of the historic forecast.  The current cycle time may not be the forecast reference time of the historic forecast.  What happens if the validity time on the coefficients cube is not the same / representative of the validity time on the historic forecasts?

Reply:
I've re-worked this method, as well as the metadata that is added to the resulting coefficients cube. 

A few key points are:
* The usage of the current cycle time has been removed, so that there is no longer a requirement for the `cycletime` argument to be provided to the CLI.
* The EMOS coefficients cube now only contains a forecast period and a `forecast_reference_time` coordinate. This metadata matches the metadata output when constructing a reliability calibration table, in order to reduce unnecessary differences between these processes. 
* The functionality within the `create_coefficients_cubelist` method has been stripped down and then factored out into separate methods to handle the different steps within the construction of the coefficients cubelist.
* The checking of the temporal coordinates, specifically that the forecast period and the hour of the forecast_reference_time coordinate match between the forecast and the coefficients cubes when applying the EMOS coefficients, has been refactored to make sure that it is checking what we want.
* The `diagnostic_standard_name` attribute that is added in `create_coefficients_cubelist` is now checked when applying the EMOS coefficients to make sure that the coefficients are applied to an appropriate forecast.